From c0ca1f474d843a071fdc908d66bbe36eac3be0c4 Mon Sep 17 00:00:00 2001
From: Petr Uzel <petr.uzel@suse.cz>
Date: Wed, 26 May 2010 14:27:07 +0200
Subject: [PATCH 2/3] parted: initialize align_type to default value

* parted/parted.c (do_align_check): Initialize align_type to
default alignment; this prevents reading of uninitialized memory
later in command_line_get_align_type()
---
 parted/parted.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parted/parted.c b/parted/parted.c
index 04d765c..830075b 100644
--- a/parted/parted.c
+++ b/parted/parted.c
@@ -2080,7 +2080,7 @@ do_align_check (PedDevice **dev)
   if (!disk)
     return 0;
 
-  enum AlignmentType align_type;
+  enum AlignmentType align_type = PA_OPTIMUM;
   PedPartition *part = NULL;
   bool aligned =
     (command_line_get_align_type (_("alignment type(min/opt)"), &align_type)
-- 
1.9.0

